include/rocksdb/env.h

```diff
@@ -540,6 +540,11 @@ class Env {
     return Status::NotSupported();
   }
 
+  // Check whether the specified path is a directory
+  virtual Status IsDirectory(const std::string& /*path*/, bool* /*is_dir*/) {
+    return Status::NotSupported();
```

Comment:
Nit: Can we add a message here too? There was a case recently where a NotSupported() error was reported in the info log, but it was hard to figure out where it was coming from.

Reply:
Done.